llama2.mojo

```diff
@@ -451,43 +453,42 @@ struct TransformerWeights:
 
     fn __init__(inout self, file_name: String, config: Config) raises:
         var bytes_read = 0
-        var f = open(file_name, "r")
-
-        # throw away config data
-        _ = f.read_bytes(NUM_CONFIG_INT * sizeof[DType.int32]())
-        bytes_read += NUM_CONFIG_INT * sizeof[DType.int32]()
+        with open(file_name, "rb") as f:
```

Comment:
I wouldn't move the whole init logic under `with` context block. It increases cyclomatic complexity and makes harder the maintenance of this part.